### ISSUE TITLE
Refutation soundness on string formula

### ISSUE THREAD

**Issue by rainoftime** [Z3Prover/z3#5030]
~~~~
(assert (<= 0 (str.len "\u16a7\u01cd\u0125\u2c7b\u0155\u014f\u020b\u16c1\u2c6d\u0211\xf6\u0245\u0224}w\xfc\u16d9\u0248\u16b3") 76))
(check-sat)
~~~~
cvc4: sat
z3 1da7522893: unsat

**Comment by zhendongsu** [comment]
I think the reason is that Z3 doesn't parse for example "\u16a7" as a unicode char (so, its length would be 5):

```
[516] % z3release small.smt2 
sat
unsat
[517] % cvc4 -q small.smt2 
sat
sat
[518] % cat small.smt2 
(assert (= (str.len "\u{0000}") 1))
(check-sat)
(reset)
(assert (= (str.len "\u0000") 1))
(check-sat)
[519] % 

```

Related to https://github.com/Z3Prover/z3/issues/5029.

**Comment by NikolajBjorner** [comment]
Fixed by 6bd02e122b3ba08fc11a13d865bc8c54d9bd0af8
